internal/server/scim.go

```diff
@@ -22,6 +25,13 @@ func ListProviderUsers(c *gin.Context, r *api.SCIMParametersRequest) (*api.ListP
 		StartIndex: r.StartIndex,
 		Count:      r.Count,
 	}
+	if r.Filter != "" {
+		exp, err := filter.ParseFilter([]byte(r.Filter))
```

Comment:
I couldn't find any fuzz tests in the scim2/filter-parser repo, so we should write some that exercise `filter.ParseFilter` There are a few examples of that for `uid.ID`.  I'd be happy to pair on that with you, if you're interested. 

I guess we mostly want to test that it doesn't panic, and that the parse finishes in a relatively short duration (effectively testing that it doesn't get stuck in some infinite loop on bad input).

Reply:
Added a fuzz test that makes sure this completes in 5 seconds and does not panic, good suggestion